### PR TITLE
Check instanceof ApiException fixed in SimpleErrorHandler

### PR DESCRIPTION
**Pull request by miranovy**
Hi,
  your commits from the last days are great :+1: 

I allowed to fix one bug in SimpleErrorHandler. Variable $error can never be of the type ApiResponse, but type of ApiException.

I also added default message to ValidationException. Original message "Application encountered an internal error. Please try again later." is not is not entirely ideal ;)

Best regards
Míra

### CODE REVIEW

File: src/Exception/Api/ValidationException.php [hunk 1]
```diff
@@ -11,7 +11,7 @@ class ValidationException extends ClientErrorException
 	/**
 	 * @param mixed[] $fields
 	 */
-	public function __construct(string $message = '', int $code = 422, ?Throwable $previous = null, array $fields = [])
+	public function __construct(string $message = 'Validation error. See details in the context.', int $code = 422, ?Throwable $previous = null, array $fields = [])
```
Comment (mabar) [inline src/Exception/Api/ValidationException.php]: Keep it in PR, but in separate commit please.

File: src/ErrorHandler/SimpleErrorHandler.php [hunk 2]
```diff
@@ -51,7 +51,7 @@ protected function createResponseFromError(Throwable $error): ResponseInterface
 		$data = [
 			'status' => 'error',
 			'code' => $code,
-			'message' => $error instanceof ApiResponse ? $error->getMessage() : 'Application encountered an internal error. Please try again later.',
+			'message' => ($error instanceof ApiException && $error->getMessage() !== '') ? $error->getMessage() : 'Application encountered an internal error. Please try again later.',
```
Comment (mabar) [inline src/ErrorHandler/SimpleErrorHandler.php]: Without message empty check please. ApiException and it descendants should always have an message. We will enforce it inside ApiException soon.

Comment (mabar) [reply src/ErrorHandler/SimpleErrorHandler.php]: Anyway, with your fix will be message `'Application encountered an internal error. Please try again later.'` displayed only for exceptions which have no meaning for api layer

File: src/Exception/Api/ValidationException.php (hunk 1 above)
Comment (miranovy) [reply src/Exception/Api/ValidationException.php]: Sorry for that :(

File: src/ErrorHandler/SimpleErrorHandler.php (hunk 2 above)
Comment (miranovy) [reply src/ErrorHandler/SimpleErrorHandler.php]: Message empty check removed.

**Comment by mabar** [comment]
Thanks

**Comment by mabar** [comment]
Btw, check `git rebase -i HEAD~n` where `n` is number of commits you want to edit. It's nice when you need modify history (remove, merge, rename commits etc)

**Comment by miranovy** [comment]
> Btw, check `git rebase -i HEAD~n` where `n` is number of commits you want to edit. It's nice when you need modify history (remove, merge, rename commits etc)

Thanks you

**Pull request merged**